doc/performance_blog_post/performance_blog_post.md

```diff
@@ -0,0 +1,120 @@
+| testLoadSalesParquetFiles          | 219.21    |
+
+The first 4 cases (`testLoadManyParquetRows`, `testLoadLargeParquetRows`, `testLoadManyParquetColumns` `testLoadManyParquetRowsFromOneFile`) all load the same amount of data (1 GB). They just differ on how the data is distributed over rows, files and columns.
+
+This dataset consists of pseudo-random generated strings. The different test cases are built as show in the following table:
+
+| Test Case                          | String size in byte | Rows      | Files | Columns |
+|------------------------------------|---------------------|-----------|-------|---------|
+| testLoadManyParquetRowsFromOneFile | 1,000               | 1,000,000 | 1     | 1       |
+| testLoadManyParquetRows            | 100                 | 1000000   | 10    | 1       |
+| testLoadLargeParquetRows           | 1,000,000           | 100       | 10    | 1       |
+| testLoadManyParquetColumns         | 1000                | 1000      | 10    | 100     |
+
+You can observe that the product of all columns is always 1,000,000,000 so 1 GB.
+
+Similar the two trailing test split the same test data in 200 (`testLoadSalesParquetFiles`) or 8 `testLoadFewBigSalesParquetFiles`. There the data is exemplary sales data consisting of 7 decimal columns with 3,621,459,710 rows.
+
+![Performance test results of Parquet file loading 1](slowParquetTestResults1.png)
+![Performance test results of Parquet file loading 2](slowParquetTestResults2.png)
+
+From the charts, it becomes clear that the Virtual Schema at that point has an issue with loading data from few big files. When loading from more files, it loads the same amount of data a lot quicker.
+
+Starting with this analysis, we discussed with our product management and decided to invest into performance improvements for loading big Parquet files. More precisely we set the goal to make improvements so that the `testLoadFewBigSalesParquetFiles` also runs in less than 250s.
+
+## Improving the Performance
+
+Next we analyzed the source code for possible reasons for the slow reading of big Parquet files. We identified the following issues:
+
+* Unequal distribution
+* Too few parallelization for few Parquet files
+
+The query processing of the Files Virtual Schemas consists of two phases: A planning phase and an execution phase. In the planning phase the adapter analyzes the query and starts multiple workers (UDFs) that then run the execution phase. For technical reasons the workers can not communicate at the moment. So the Virtual Schema parallelizes the import job over multiple workers.
+
+### Distributing the File
+
+Until now this parallelization was realized using hash binning. So each worker received a range and then loaded files that names hash values fell into that range. That approach is great for distributing lots of files, since it does not require to list all files in advance. Doing so could cause memory overflow when distributing for example 1,000,000 files with long names.
```

Comment:
```suggestion
Until now this parallelization was realized using hash binning. So each worker received a range and then loaded files that names hash values fell into that range. That approach is great for distributing lots of files since it does not require to listing files in advance. Doing so could cause memory overflow when distributing for example 1,000,000 files with long names.
```